connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/LoggingResource.java

```diff
@@ -158,43 +157,50 @@ public Response setLevel(final @PathParam("logger") String namedLogger,
         List<String> modifiedLoggerNames = new ArrayList<>();
         for (Logger logger: childLoggers) {
             logger.setLevel(level);
-            modifiedLoggerNames.add(logger.getName());
+            if ("".equals(logger.getName())) {
+                modifiedLoggerNames.add("root");
+            } else {
+                modifiedLoggerNames.add(logger.getName());
+            }
         }
         Collections.sort(modifiedLoggerNames);
 
         return Response.ok(modifiedLoggerNames).build();
     }
 
-    protected Logger lookupLogger(String namedLogger) {
-        return LogManager.getLogger(namedLogger);
+    protected Logger lookupLogger(String loggerName) {
+        final LoggerContext loggerContext = (LoggerContext) LogManager.getContext(false);
+
+        return loggerContext.getLogger(loggerName);
     }
 
     @SuppressWarnings("unchecked")
-    protected Enumeration<Logger> currentLoggers() {
-        return LogManager.getCurrentLoggers();
+    protected List<Logger> currentLoggers() {
+        final LoggerContext loggerContext = (LoggerContext) LogManager.getContext(false);
+
+        return loggerContext.getLoggers()
+            .stream()
+            .filter(logger -> !logger.getName().equals(""))
```

Comment:
This is the second time you've got a `.equals("")`. It might be worth factoring into a `isRootLogger(Logger)` method.